fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkcommon/BulkDataUtils.java

```diff
@@ -155,41 +156,89 @@ private static int getFhirResourceFromBufferReader(BufferedReader resReader, int
+            boolean isReuseInput, ImportTransientUserData transientUserData) {
+        int exported = 0;
+        if (isReuseInput) {
+            try {
+                if (transientUserData.getBufferReader() == null) {
+                    BufferedReader resReader = Files.newBufferedReader(Paths.get(filePath));
+                    transientUserData.setBufferReader(resReader);
+                }
+                exported = getFhirResourceFromBufferReader(transientUserData.getBufferReader(), 0, fhirResources);
+            } catch (Exception ioe) {
+                logger.warning("readFhirResourceFromLocalFile: " + "Error proccesing file " + filePath + " - " + ioe.getMessage());
+                exported = 0;
+            }
+        } else {
+            try (BufferedReader resReader = Files.newBufferedReader(Paths.get(filePath))) {
+                exported = getFhirResourceFromBufferReader(resReader, numOfLinesToSkip, fhirResources);
+            } catch (Exception ioe) {
+                logger.warning("readFhirResourceFromLocalFile: " + "Error proccesing file " + filePath + " - " + ioe.getMessage());
+                exported = 0;
+            }
         }
         return exported;
     }
 
 
-    public static int readFhirResourceFromHttps(String dataUrl, int numOfLinesToSkip, List<Resource> fhirResources) {
-        int exported;
-        try (BufferedReader resReader = new BufferedReader(new InputStreamReader(new URL(dataUrl).openConnection().getInputStream()))) {
-            exported = getFhirResourceFromBufferReader(resReader, numOfLinesToSkip, fhirResources);
-        } catch (Exception ioe) {
-            logger.warning("readFhirResourceFromHttps: " + "Error proccesing file " + dataUrl + " - " + ioe.getMessage());
-            exported = 0;
+    public static int readFhirResourceFromHttps(String dataUrl, int numOfLinesToSkip, List<Resource> fhirResources,
+            boolean isReuseInput, ImportTransientUserData transientUserData) {
+        int exported = 0;
+        if (isReuseInput) {
+            try {
+                if (transientUserData.getBufferReader() == null) {
+                    InputStream inputStream = new URL(dataUrl).openConnection().getInputStream();
+                    transientUserData.setInputStream(inputStream);
+                    BufferedReader resReader = new BufferedReader(new InputStreamReader(inputStream));
+                    transientUserData.setBufferReader(resReader);
+                }
+                exported = getFhirResourceFromBufferReader(transientUserData.getBufferReader(), 0, fhirResources);
+            } catch (Exception ioe) {
+                logger.warning("readFhirResourceFromHttps: " + "Error proccesing file " + dataUrl + " - " + ioe.getMessage());
+                exported = 0;
+            }
+        } else {
+            try (BufferedReader resReader = new BufferedReader(new InputStreamReader(new URL(dataUrl).openConnection().getInputStream()))) {
```

Comment:
226 uses a non autocloseable. here we use an autocloseable --- worth doublechecking

Reply:
yeah,  I did this on purpose. if the stream/reader is reused across all the chunks, then they will only be closed at the end of the processing of this data source.